lib/ansible/module_utils/facts/timeout.py

```diff
@@ -25,29 +26,30 @@
 DEFAULT_GATHER_TIMEOUT = 10
 
 
-class TimeoutError(Exception):
-    pass
+try:
+    # On Python 3, reuse the TimeoutError
+    class TimeoutError(TimeoutError):
+        pass
+except NameError:
+    # On Python 2, there unfortunately isn't a better base class to inherit from
+    class TimeoutError(Exception):
```

Comment:
in py3 TimeoutError inherits from OSError, making this code subject to divergent paths in py2 vs py3, so if you really want to avoid that inherit from OSError, not Exception

Reply:
Will do